### PR TITLE
Add barebones ninjascript transpiler

### PR DESCRIPTION
**Pull request by sigvef**
Now we can edit src/MyLayer.n instead of src/MyLayer.js, and
src/MyLayer.js will be automatically generated. Only an underspecified
subset of ninjascript is implemented.

### CODE REVIEW

**Comment by stianjensen** [comment]
Okay, now we're talking

File: nin/backend/tests/TestLayer.n [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+this.someMemberVariable = "hello world"
+
+someMethod(a, b, c) {
+    var scopedVariable = "someMethod was called"
+        log(scopedVariable, a, b, c)
+}
+
+someMethodThatReturnsSomething(a, b) {
+    < a * b
```
Comment (stianjensen) [inline nin/backend/tests/TestLayer.n]: Yeah, I don't like this syntax anymore :(

**Comment by stianjensen** [comment]
Okay, so this is pretty neat.

Although, most of these syntax-slimming features are pretty redundant or minimal when ES6 class syntax comes into the mainstream. So I'm not sure if we actually want to complicate nin further by merging it.

**Comment by sigvef** [comment]
> Okay, now we're talking
> 
> EDIT: and you've kept it a secret all this time!

I totally forgot that it even existed :P

File: nin/backend/tests/TestLayer.n (hunk 1 above)
Comment (sigvef) [inline nin/backend/tests/TestLayer.n]: Haha, me neither. Arbitrary cryptic symbols suck!

**Comment by sigvef** [comment]
> Although, most of these syntax-slimming features are pretty redundant or minimal when ES6 class syntax comes into the mainstream. So I'm not sure if we actually want to complicate nin further by merging it.

I agree. Ninjascript doesn't actually bring much to the table. We should probably rather just start using es6-style classes and reevaluate the need for a new language later.

**Pull request closed without merging**